### PR TITLE
TEC-4300: fix components and new component BoxSelect

### PR DESCRIPTION
**Pull request by samuelvermeuln**
## Task link
[JIRA](https://evinobr.atlassian.net/jira/software/projects/TEC/boards/64?selectedIssue=TEC-4300&text=%5Bfront%5D)

## Task description
fix components and new component BoxSelect

### CODE REVIEW

File: src/components/Select/SelectBox.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  isClosed,
+  ...selectProps
+}) => {
+  const hasValue = Boolean(selectProps.value)
+
+  return (
+    <SelectContainer theme={theme} hasValue={hasValue}>
+      {iconLeft && <IconLeft>{iconLeft}</IconLeft>}
+      <SelectCustom iconLeft={iconLeft} {...selectProps}>
+        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```
Comment (joelsalmeida) [inline src/components/Select/SelectBox.tsx]: This eslit-disabled shouldn't be here. Check this reference on how to make the association correctly: [The HTML Select element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#try_it)

File: src/components/Select/SelectBox.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import React from 'react'
```
Comment (joelsalmeida) [inline src/components/Select/SelectBox.tsx]: This component is non-standard. "Folder structure, css files and stories..."

File: src/components/Select/Select.stories.tsx [hunk 3]
```diff
@@ -191,3 +194,33 @@ export const Responsible: Story = {
     </Select>
   ),
 }
+
+const TemplateSelect = () => {
```
Comment (joelsalmeida) [inline src/components/Select/Select.stories.tsx]: Each component should have its own stories. "With "meta", arguments and stories..."

File: src/components/Select/Select.stories.tsx [hunk 4]
```diff
@@ -191,3 +194,33 @@ export const Responsible: Story = {
...
+  const [selectedValue, setSelectedValue] = useState('')
```
Comment (joelsalmeida) [inline src/components/Select/Select.stories.tsx]: Why this state?

File: src/components/Input/index.tsx [hunk 5]
```diff
@@ -1,20 +1,18 @@
...
 
 interface InputProps extends HTMLAttributes<HTMLInputElement> {
-  label?: string
-  value?: string
-  icon?: React.ReactNode
-  theme?: ThemeType
-  type?: string
   placeholder?: string
+  theme?: ThemeType
+  iconLeft?: React.ReactNode
```
Comment (joelsalmeida) [inline src/components/Input/index.tsx]: There is no story for this cases. (iconLeft and iconRight)

Comment (joelsalmeida) [reply src/components/Input/index.tsx]: Furthermore... the controls in the documentation are not working.

**Review by joelsalmeida — CHANGES_REQUESTED** [review]

File: src/components/Select/SelectBox.tsx (hunk 1 above)
Comment (samuelvermeuln) [reply src/components/Select/SelectBox.tsx]: this guy asks for a <label/> around...
and here we don't pass
then eslint complains

File: src/components/Select/Select.stories.tsx (hunk 3 above)
Comment (samuelvermeuln) [reply src/components/Select/Select.stories.tsx]: You will have to delete the previous select to use this one. and then create all the stories
I made this one faster to use in the form task

File: src/components/Select/Select.stories.tsx (hunk 4 above)
Comment (samuelvermeuln) [reply src/components/Select/Select.stories.tsx]: and the value of the select so it doesn't remain empty on the story-book screen

File: src/components/Select/Select.stories.tsx [hunk 6]
```diff
@@ -1,32 +1,16 @@
 import type { Meta, StoryObj } from '@storybook/react'
-import styled from 'styled-components'
-import { MINIMAL_VIEWPORTS } from '@storybook/addon-viewport'
+import React, { useState } from 'react'
+import { ChevronDown, Close, Search } from 'semillon'
```
Comment (joelsalmeida) [inline src/components/Select/Select.stories.tsx]: "Search" is not being used.

File: src/components/Select/Select.stories.tsx [hunk 7]
```diff
@@ -40,154 +24,84 @@ const meta: Meta<typeof Select> = {
...
-
-export default meta
+} as Meta
 
 //* STORIES */
 
-type Story = StoryObj<typeof Select>
+type Story = StoryObj<SelectProps>
+
+const TemplateSelect = (props) => {
```
Comment (joelsalmeida) [inline src/components/Select/Select.stories.tsx]: Type this property.

File: src/components/Select/index.tsx [hunk 8]
```diff
@@ -1,27 +1,53 @@
...
+      hasValue={hasValue}
+      error={error}
+      theme={theme}
+      disabled={disabled}
     >
-      {children}
-    </SelectStyled>
+      <IconLeft>{iconLeft}</IconLeft>
+      <SelectCustom disabled={disabled} ref={selectRef} iconLeft={iconLeft} {...selectProps}>
+        <option value="" />
```
Comment (joelsalmeida) [inline src/components/Select/index.tsx:40]: Why this empty option here?

File: src/components/Input/types.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React, { HTMLAttributes } from 'react'
+import { ThemeType } from '../../types'
+
+export interface InputProps extends HTMLAttributes<HTMLInputElement> {
```
Comment (joelsalmeida) [inline src/components/Input/types.ts:4]: I think it is unnecessary to add "undefined" type for non-mandatory properties and error should just be boolean.

File: src/components/Select/index.tsx [hunk 10]
```diff
@@ -1,27 +1,53 @@
...
+import {
+  SelectContainer,
+  SelectCustom,
+  Label,
+  IconLeft,
+  IconRight,
+} from './style'
+import { SelectProps, SelectRef } from './types'
+
+/* eslint-disable */
```
Comment (joelsalmeida) [inline src/components/Select/index.tsx:12]: Why this eslint-disable?

File: src/components/Select/style.tsx [hunk 11]
```diff
@@ -1,50 +1,122 @@
...
+
+      ${Label} {
+        color: ${Colors[theme].feedback.feedbackError100};
+      }
+    `}
+
+  ${({ disabled, theme }) =>
+    disabled &&
+    css`
+      border-color: ${Colors[theme].neutral.neutral200} !important;
```
Comment (joelsalmeida) [inline src/components/Select/style.tsx:68]: Type errors on line 68 and 71.

File: src/components/Input/Input.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
+import { Meta, StoryObj } from '@storybook/react'
+import { Search, Close } from 'semillon'
+import React from 'react'
+import { Input } from '../index'
+import { InputProps } from './types'
+
+export default {
```
Comment (joelsalmeida) [inline src/components/Input/Input.stories.tsx:7]: Type errors here.

File: src/components/Select/index.tsx (hunk 8 above)
Comment (samuelvermeuln) [reply src/components/Select/index.tsx:40]: make the effect that is in figma

File: src/components/Select/index.tsx (hunk 10 above)
Comment (samuelvermeuln) [reply src/components/Select/index.tsx:12]: It is necessary, delete the line and you will understand

**Review by joelsalmeida — APPROVED** [review]

**Pull request merged**